File: Tp POO/Ejercicio2/Alumno.py
from Nota import  Nota
class Alumno:
    def __init__(self, nombreCompleto, legajo):
        self.nombreCompleto = nombreCompleto
        self.legajo = legajo
        self.notas = []

    def ingresoNota(self,notas):
        self.notas = notas
    
    def __str__(self):
        return  f"---\nNombre: {self.nombreCompleto},\nLegajo: {self.legajo},\nNotas: {self.notas}\n---"

File: Programacion 1/Python/Tps lab Java/src/Asociaciones/CargaNotas/Alumno.java
package Asociaciones.CargaNotas;

import java.util.ArrayList;

public class Alumno {
    private String nombre;
    private long legajo;
    private ArrayList<Nota> notas;

    public Alumno(String nombre, long legajo) {
        this.nombre = nombre;
        this.legajo = legajo;
        this.notas = new ArrayList<>();
    }

    public void agregarNota(Nota nota) {
        notas.add(nota);
    }

    public double calcularPromedio(){
        double promedio = 0;
        for (Nota nota : notas) {
            promedio += nota.getNotaExamen();
        }
        return promedio / notas.size();
    }

    @Override
    public String toString() {
        return "Alumno: " +
                "\nNombre = '" + nombre + '\'' +
                "\nLegajo = " + legajo +
                "\nNotas = " + notas +
                "\n------" +
                "\nPromedio = "+ calcularPromedio();
    }
}
